### ISSUE TITLE
Explore using SVG for page rendering in editor

### ISSUE THREAD

**Issue by barklund** [GoogleForCreators/web-stories-wp#5710]
## Feature Description

Explore using SVG's in these places:

* Canvas
* Thumbnail
* Text sets
* Page layouts
* Dashboard story preview

This will also greatly simplify coordinate calculations currently made complicated by having to calculate in and out of different coordinate spaces (which might not be fully resolved by this, but somewhat).

This is an exploratory ticket with the goal of an engineering document outlining the options, problems, advantages and, roadblocks.

**Comment by swissspidy** [comment]
> * Canvas

Did you mean Carousel?

Also: (how) does the eyedropper fit into this?

**Comment by maxyinger** [comment]
I explored this a little bit for speeding up rendering story content. I think the biggest challenge with perf right now in Text sets, Page layouts & Dashboard story preview is just pure amount of nodes. Here are some high level takeaways I found with svg:

**PROS**
- Less coordinate translation work cus everything is relative to the viewbox (like you mentioned).
- I believe this approach could actually handle our shape masking nicely
- If we created logic to generate an svg from story data we could encode it as a dataURI, which would allow us to only have to render once, then we could cache the results or offload this logic on a server which would be huge for perf

**CONS**
- If we're still doing all the react rendering, just with the end result being svg instead of html, I think we still run into the main bottleneck of rendering so many nodes being the primary hamper on perf
- As far as I looked into it, doing any encoding of an svg would break any non-system fonts being used by text within the svg
- Would break animations cus svg coords if inline (and if not inline, can't reach into the svg and animate anything). Although a solution to this I thought of is just lazy loading in actual html story rendering as needed when we need to animate, and using the generated svg image as a placeholders for story pages

I do think our focus should be geared towards generating some sort of image (rendering once), then passing that around vs rendering every node all the time and speeding up the rendering perf of each node (I think we still run into the same bottlenecks here).

Anyway, these are just the conclusions I came to in exploring this briefly. Definitely think another exploratory perf audit could benefit us here even if it just solidifies my earlier findings.

Also if anybody sees possible solutions to any of the cons listed above I'd love to hear them. That's just as far as I got without making this a huge time suck.

**Ideas**
One idea I had to to combat that font issue could be to render the svg somewhere where the fonts it's using are loaded, then paint that to a canvas with drawImage, then extract a data url from that canvas (basically flatten/rasterize the svg).
